src/apps/js_generic/js_generic_base.cpp

```diff
@@ -220,9 +220,9 @@ namespace ccfapp
       }
       request.set("params", params);
 
-      const auto& request_body = endpoint_ctx.rpc_ctx->get_request_body();
+      auto request_body = new std::vector<uint8_t>(endpoint_ctx.rpc_ctx->get_request_body());
```

Comment:
Why the copy? Even assuming JS_SetOpaque takes ownership, it's not great to have another call in between that could throw.

Reply:
As discussed, have updated this to remove the new `new`s. The `body` will always refer to the _current_ request, and the `map_name` is stored as a property (`_map_name`) rather than an `Opaque`, so it should be cleaned up by the normal QuickJS flow rather than a finalizer.